edb/testbase/server.py

```diff
@@ -289,6 +292,36 @@ async def connect(cls, *,
             cluster=cluster, database=database, user=user, password=password)
         return await edgedb.async_connect(**conargs)
 
+    def repl(self):
+        """Open interactive EdgeQL REPL right in the test.
+
+        This is obviously only for debugging purposes.  Just add
+        `self.repl()` at any point in your test.
+        """
+
+        conargs = self.get_connect_args()
+
+        cmd = [
+            # TODO: switch to 'edgedb' when it understands EDGEDB_PASSWORD
+            'python', '-m', 'edb.cli',
+            '--host', conargs['host'],
+            '--port', str(conargs['port']),
+            '--database', self.con.dbname,
+            '--user', conargs['user'],
+        ]
+
+        env = os.environ.copy()
+        if password := conargs.get('password'):
+            env['EDGEDB_PASSWORD'] = password
```

Comment:
Both repls support `--password-from-stdin`, could you use that?

Reply:
No, because you need to attach repl process' stdin to the main process' stdin.